vortex-scalar/src/arrow.rs

```diff
@@ -130,3 +130,18 @@ impl TryFrom<&Scalar> for Arc<dyn Datum> {
         }
     }
 }
+
+impl Scalar {
+    pub fn to_varbin_datum(self) -> VortexResult<Arc<dyn Datum>> {
+        match self.dtype {
+            DType::Utf8(_) => {
+                value_to_arrow_scalar!(self.value.as_buffer_string()?, StringArray)
+            }
+            DType::Binary(_) => {
+                value_to_arrow_scalar!(self.value.as_buffer()?, BinaryArray)
+            }
+
+            other => vortex_bail!("Expected Utf8 or Binary scalar, found {other}"),
+        }
+    }
+}
```

Comment:
This is a gross thing for Arrow. Basically we want the `From<Scalar> for Datum` impl to assume StringView but we need another way to create a non-view scalar

Reply:
Actually this should probably live in the varbin mod